networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/gossip/GossipWireValidator.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.networking.p2p.libp2p.gossip;
+
+import io.libp2p.pubsub.PubsubMessageValidator;
+import org.jetbrains.annotations.NotNull;
+import pubsub.pb.Rpc;
+import pubsub.pb.Rpc.RPC;
+
+/**
+ * Validates Gossip messages at the level of Protobuf structures Rejects messages with prohibited
+ * Gossip fields: {@code from, signature, seqno}
+ */
+public class GossipWireValidator implements PubsubMessageValidator {
+
+  public static final GossipWireValidator INSTANCE = new GossipWireValidator();
```

Comment:
Any particular reason to make this a singleton?  Its not very common in the teku codebase.

Reply:
I remember you had the similar concern regarding very similar case (found it: https://github.com/PegaSysEng/teku/pull/2376#discussion_r457587546). I'm not sure I understand your concerns regarding this pattern. Can you please explain in more details? From my perspective singleton is suitable in this case cause it has no state and just a single class instance makes sense here. 

Do you think this variant is better: https://github.com/PegaSysEng/teku/pull/2960/commits/141ddf1e4610b1a2429a6883bfe392b29cb620c1 ?

Reply:
In general, singletons tend to be a bit controversial because they create global state and can make testing complicated in terms of being able to inject unique instances / mock things easily.